### PR TITLE
Update interop packages to build for dev17

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Microsoft.VisualStudio.InteractiveWindow.Tests/InteractiveWindowTests.cs [hunk 1]
```diff
@@ -44,11 +43,6 @@ void IDisposable.Dispose()
 
         private IInteractiveWindow Window => _testHost.Window;
 
-        private Task TaskRun(Action action)
```
Comment (jmarolf) [inline src/Microsoft.VisualStudio.InteractiveWindow.Tests/InteractiveWindowTests.cs:47]: Why this change?

File: src/Microsoft.VisualStudio.InteractiveWindow.Tests/TestUtilities/Threading/ForegroundThreadAffinitizedObject.cs [hunk 2]
```diff
@@ -33,12 +28,7 @@ internal static ForegroundThreadData CreateDefault(ForegroundThreadDataKind defa
         {
             var kind = ForegroundThreadDataInfo.CreateDefault(defaultKind);
 
-            // None of the work posted to the foregroundTaskScheduler should block pending keyboard/mouse input from the user.
-            // So instead of using the default priority which is above user input, we use Background priority which is 1 level
-            // below user input.
-            var taskScheduler = new SynchronizationContextTaskScheduler(new DispatcherSynchronizationContext(Dispatcher.CurrentDispatcher, DispatcherPriority.Background));
```
Comment (jmarolf) [inline src/Microsoft.VisualStudio.InteractiveWindow.Tests/TestUtilities/Threading/ForegroundThreadAffinitizedObject.cs:39]: why this change?

**Comment by jmarolf** [comment]
This PR appears to add new APIs and clean up threading logic both of which should probably be done in a more targeted change.

File: src/Microsoft.VisualStudio.InteractiveWindow.Tests/InteractiveWindowTests.cs (hunk 1 above)
Comment (sharwell) [reply src/Microsoft.VisualStudio.InteractiveWindow.Tests/InteractiveWindowTests.cs:47]: A warning was reported for the name of `TaskRun`. Replacing with `Task.Run` seemed to be simpler than renaming once here only to remove it in the future.

**Comment by sharwell** [comment]
> This PR appears to add new APIs

No new APIs are added. The old analyzer failed to recognize APIs we already had.

> and clean up threading logic

Threading logic changes were constrained to the test project. Suppressions were used for all but the most obvious cases in production code. We can still refactor this but we have a hard deadline that this be published to the internal feeds _today_.

**Review by jmarolf — APPROVED** [review]

File: src/Microsoft.VisualStudio.InteractiveWindow.Tests/InteractiveWindowTests.cs (hunk 1 above)
Comment (sharwell) [reply src/Microsoft.VisualStudio.InteractiveWindow.Tests/InteractiveWindowTests.cs]: Removed this change and used a suppression instead.

File: src/Microsoft.VisualStudio.InteractiveWindow.Tests/TestUtilities/Threading/ForegroundThreadAffinitizedObject.cs (hunk 2 above)
Comment (sharwell) [reply src/Microsoft.VisualStudio.InteractiveWindow.Tests/TestUtilities/Threading/ForegroundThreadAffinitizedObject.cs]: Removed this change and used a bunch of suppressions instead.

**Comment by genlu** [comment]
@sharwell @tmat mentioned that we can use main for dev17, since it's unlikely we will need to insert again for dev16.

File: eng/Versions.props [hunk 3]
```diff
@@ -10,40 +10,37 @@
     <UsingToolSymbolUploader>true</UsingToolSymbolUploader>
```
Comment (tmat) [inline eng/Versions.props:10]: Let's update `VersionPrefix` to `4.0.0` since this is Dev17.

Comment (sharwell) [reply eng/Versions.props:10]: Updated this and the vsixmanifest

**Review by tmat — APPROVED** [review]
:shipit:
